rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -186,6 +183,14 @@ class TensorInfo {
     bool is_image() const { return _is_image; }
     void set_metadata() { _is_metadata = true; }
     bool is_metadata() const { return _is_metadata; }
+    void set_roi_ptr(unsigned *roi_ptr) { 
+        auto deleter = [&](unsigned *ptr) {};   // Empty destructor used, since memory is handled by the pipeline
+        _roi.reset(roi_ptr, deleter); 
+    }
+    void copy_roi(void *roi_buffer) {
+        if(_roi != nullptr && roi_buffer != nullptr)
+            memcpy((void *)roi_buffer, (const void *)_roi.get(), _batch_size * sizeof(RocalROI));
```

Comment:
We are using roi & roi_buffer.
Whats the usecase of these ?

We would copy the roi to roi_buffer, why cant we directly return the roi?